src/category/category.service.ts

```diff
@@ -52,7 +62,7 @@ export class CategoryService {
     if (!found) {
       throw new NotFoundException(`${Message.NOT_FOUND.cateogory}=${id}`);
     } else {
-      await this.categoryRepository.delete({ id });
+      await this.categoryRepository.save({ ...found, isDelete: true });
```

Comment:
Return the entity that was deleted or return nothing.

Reply:
done